agent/resources/cgroup/cgroup_controller_linux.go

```diff
@@ -0,0 +1,73 @@
+// +build linux
+
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cgroup
+
+import (
+	"github.com/cihub/seelog"
+	"github.com/containerd/cgroups"
+	"github.com/pkg/errors"
+)
+
+var factory CgroupFactory = &GlobalCgroupFactory{}
```

Comment:
I don't have a different suggestion yet, but package-level variables like this are an anti-pattern I'd like to avoid as much as possible.

If you're only doing this to make testing possible, I think I can live with it but it needs a clear indication that it's only for tests.